pkg/services/alerting/rule.go

```diff
@@ -140,7 +140,13 @@ func NewRuleFromDBAlert(ruleDef *models.Alert) (*Rule, error) {
 		if id, err := jsonModel.Get("id").Int64(); err == nil {
 			uid, err := translateNotificationIDToUID(id, ruleDef.OrgId)
 			if err != nil {
-				logger.Error("Unable to translate notification id to uid", "error", err.Error(), "dashboardId", model.DashboardID, "alertId", model.ID, "panelId", model.PanelID, "notificationId", id)
+				if !errors.Is(err, models.ErrAlertNotificationFailedTranslateUniqueID) {
+					return nil, ValidationError{Reason: "Failed to tranlate notification id to uid:" + err.Error(), DashboardID: model.DashboardID, AlertID: model.ID, PanelID: model.PanelID}
```

Comment:
I'm not 100% sure here if in case of an actual database problem we want to cause the alert to fail or just log the error.

Reply:
So if this error is returned it would end up here when reading alert rules from the database :
https://github.com/grafana/grafana/blob/c8658f3ee8408d3e7db7615740a30d3d5cfc1f90/pkg/services/alerting/reader.go#L40

That's a good question actually.The engine will continue to call `fetch()`, as seen below, so on one hand it will be "retried" on the other hand it may take a while for it to be "retried":
https://github.com/grafana/grafana/blob/c8658f3ee8408d3e7db7615740a30d3d5cfc1f90/pkg/services/alerting/engine.go#L85

Given there's a cache as well in sqlstore and having notifier id instead of uid should be less likely I think we want to log the error instead of returning it. WDYT?

Reply:
I agree that logging the error is preferable; I'll change it.